src/GraphQL/Types/Composite/AutoRegisteringInputObjectGraphType.cs

```diff
@@ -17,6 +17,8 @@ namespace GraphQL.Types
     /// <typeparam name="TSourceType"></typeparam>
     public class AutoRegisteringInputObjectGraphType<TSourceType> : InputObjectGraphType<TSourceType>
     {
+        private readonly Expression<Func<TSourceType, object?>>[]? _excludedProperties;
```

Comment:
Unused after `GetRegisteredProperties` call.

Reply:
That is after ctor called.

Reply:
Correct.  No way around this without a poorer OOP design.

Reply:
Probably because of this I made single method.

Reply:
```csharp
public AutoRegisteringInputObjectGraphType(params Expression<Func<TSourceType, object?>>[]? excludedProperties)
        {
            _excludedProperties = excludedProperties;
            AutoRegisteringHelper.SetFields(this, GetRegisteredProperties());
           _excludedProperties  = null;
        }
```

Reply:
make sense.

Reply:
well, then constructors of derived classes would not return the same data for `GetRegisteredProperties`.  I would leave it as is.

Reply:
Holding a few initialization properties in memory should not matter.

Reply:
Why should they be kept if you can pass them into the method? If someone needs to keep them, then it will save them in the ctor of derived class, but in general `GetRegisteredProperties` method is called once.

Reply:
This special constructor as a whole is a patch. It should not change a good design to be specific for this use case.

Reply:
And, the design needs to be considered one more time.  A derived class's constructor cannot be passed arguments that change the behavior of methods in the class.  Probably nothing we can do about it, since there is no `Initialize` call to an `IGraphType` before it is used.

Reply:
I thought about this some more and will push another change.